packages/app-explorer/src/BlockInfo/Extrinsics.tsx

```diff
@@ -40,8 +41,14 @@ class Extrinsics extends React.PureComponent<Props> {
 
   // FIXME This is _very_ similar to what we have in democracy/Item
   private renderExtrinsic = (extrinsic: Extrinsic, index: number) => {
+    const { blockNumber, t } = this.props;
     const { meta, method, section } = Method.findFunction(extrinsic.callIndex);
 
+    let eraEnd;
+    if (extrinsic.signature.era.isMortalEra) {
```

Comment:
Not convinced. So the block it shows, it the block it has been included at. So basically, you can send at `#1234` it gets included at `#1239` and now we will show `#1239`, instead of the initial value.

Decode it, and then use the period and phase - https://github.com/polkadot-js/api/blob/master/packages/types/src/type/ExtrinsicEra.ts#L192

Reply:
I was wrong. Looked with @kwingram25 through the code, it seems to be doing the right thing.